### PR TITLE
Mobx store for organization

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

Extract business logic tied to `Organization` to mobx, so the logic is testable and not tied to the React component.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: projects/cloud/app/frontend/stores/HomeStore.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  async load(organizationName: string) {
+    await this.userStore.load();
+    await this.organizationStore.load(organizationName);
+  }
+}
+
+// @ts-ignore
+export const HomeStoreContext = createContext<HomeStore>();
```
Comment (pepicrft) [inline projects/cloud/app/frontend/stores/HomeStore.ts:25]: Have you considered using [this integration](https://mobx.js.org/react-integration.html) instead?

File: projects/cloud/app/frontend/stores/OrganizationStore.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+} from '../graphql/types';
+
+class OrganizationStore {
+  organization: OrganizationQuery['organization'];
+
+  client: ApolloClient<object>;
+
+  constructor(client: ApolloClient<object>) {
+    this.client = client;
+    makeAutoObservable(this);
```
Comment (pepicrft) [inline projects/cloud/app/frontend/stores/OrganizationStore.ts:17]: Does the position of `makeAutoObservable` in the constructor matter? I've seen `HomeStore` has this one as the first line.

File: projects/cloud/app/frontend/stores/UserStore.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export default class UsersStore {
+  me: MeQuery['me'];
+  client: ApolloClient<object>;
+  constructor(client: ApolloClient<object>) {
+    this.client = client;
+    makeAutoObservable(this);
+  }
+
+  async load() {
+    const { data } = await this.client.query({
```
Comment (pepicrft) [inline projects/cloud/app/frontend/stores/UserStore.ts:14]: How are errors propagated here?

File: projects/cloud/app/frontend/components/Home.tsx [hunk 4]
```diff
@@ -270,20 +278,34 @@ const Home = () => {
...
+  const client = useApolloClient();
+  const [homeStore] = useState(new HomeStore(client));
+
+  useEffect(() => {
+    if (!organizationName) {
+      return;
+    }
+    homeStore.load(organizationName);
+  }, [organizationName]);
+
```
Comment (pepicrft) [inline projects/cloud/app/frontend/components/Home.tsx:292]: I'd consider using the [React Mobx](https://mobx.js.org/react-integration.html) integration that will take care of binding the component to observable state in the store.

**Review by pepicrft — APPROVED** [review]
Amazing work @fortmarek 👏🏼 👏🏼

File: projects/cloud/app/frontend/stores/HomeStore.ts (hunk 1 above)
Comment (fortmarek) [reply projects/cloud/app/frontend/stores/HomeStore.ts:25]: what exactly do you mean here? I am using `observer` for components but I did not want to pass the store props all over the place as context is more convenient.

File: projects/cloud/app/frontend/stores/OrganizationStore.ts (hunk 2 above)
Comment (fortmarek) [reply projects/cloud/app/frontend/stores/OrganizationStore.ts:17]: the docs show it on the [first](https://mobx.js.org/observable-state.html#makeautoobservable) line, so I'd default to that convention - although I am not sure if the result will be different. Will change it, though 👍

File: projects/cloud/app/frontend/stores/UserStore.ts (hunk 3 above)
Comment (fortmarek) [reply projects/cloud/app/frontend/stores/UserStore.ts:14]: Errors are silenced here. I could destructure the `errors` propety and re-throw it, so it's eventually caught by `ErrorBoundary`.

File: projects/cloud/app/frontend/components/Home.tsx (hunk 4 above)
Comment (fortmarek) [reply projects/cloud/app/frontend/components/Home.tsx:292]: this is the other way around - we need to reload store when the state of the component changes. In this case, the state is the URL from which we get the `organizationName`. I'm not sure if there's a cleaner way to achieve this.

**Pull request merged**